testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxrs/cfg/ResteasyAttributeTestCase.java

```diff
@@ -0,0 +1,475 @@
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.jaxrs.JaxrsAttribute;
+import org.jboss.as.test.integration.jaxrs.packaging.war.WebXml;
+import org.jboss.as.test.integration.management.base.AbstractMgmtServerSetupTask;
+import org.jboss.as.test.integration.management.base.ContainerResourceMgmtTestBase;
+import org.jboss.as.test.integration.management.util.MgmtOperationException;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.dmr.Property;
+import org.jboss.dmr.ValueExpression;
+import org.jboss.resteasy.plugins.providers.FileProvider;
+import org.jboss.resteasy.plugins.providers.SourceProvider;
+import org.jboss.resteasy.plugins.providers.StringTextStar;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.staxmapper.XMLElementReader;
+import org.jboss.staxmapper.XMLElementWriter;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests for setting RESTEasy context parameters using the Wildfly management model.
+ *
+ * @author <a href="rsigal@redhat.com">Ron Sigal</a>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(ResteasyAttributeTestCase.AttributeTestCaseDeploymentSetup.class)
+public class ResteasyAttributeTestCase extends ContainerResourceMgmtTestBase {
+
+    private static ModelControllerClient modelControllerClient;
+    private static final Map<AttributeDefinition, ModelNode> expectedValues = new HashMap<AttributeDefinition, ModelNode>();
+    private static final Map<ModelType, Class<?>> modelTypeMap = new HashMap<ModelType, Class<?>>();
+    private static Client jaxrsClient;
+
+    @ArquillianResource
+    private static URL url;
+
+    static {
+        modelTypeMap.put(ModelType.BOOLEAN, Boolean.class);
+        modelTypeMap.put(ModelType.INT, Integer.class);
+        modelTypeMap.put(ModelType.STRING, String.class);
+        modelTypeMap.put(ModelType.LIST, String.class);
+        modelTypeMap.put(ModelType.OBJECT, String.class);
+    }
+
+    private static final ModelNode ADDRESS = Operations.createAddress("subsystem", "jaxrs");
+    private static final String OUTCOME = "outcome";
+    private static final String SUCCESS = "success";
+    private static final String FAILED = "failed";
+    private static final String RESTEASY_MEDIA_TYPE_PARAM_MAPPING_CONTEXT_VALUE = "resteasy.media.type.param.mapping.context.value";
+
+    private static final ModelNode VALUE_EXPRESSION_BOOLEAN_TRUE = new ModelNode(new ValueExpression("${rest.test.true:fail}"));
```

Comment:
This is ok, but your test will be a lot simpler if you just use the following string for the ValueExpression:
```
"${org.rest.test.dummy:true}"
```
Then you don't have to worry about setting all the system properties, instead it will fall back to the default value specified in the expression. The expression resolution mechanism is well tested elsewhere, so there is no need to repeat that here. The main thing is to make sure that the runtime stuff gets the proper values.

Reply:
Nice. Ok, thanks.